node/cmd/guardiand/node.go

```diff
@@ -1818,10 +1596,49 @@ func shouldStart(rpc *string) bool {
 	return *rpc != "" && *rpc != "none"
 }
 
-func unsafeDevModeEvmContractAddress(contractAddr string) string {
-	if contractAddr != "" {
-		return contractAddr
+// checkEvmArgs verifies that the RPC and contract address parameters for an EVM chain make sense, given the environment.
+// If we are in devnet mode and the contract address is not specified, it returns the deterministic one for tilt.
+func checkEvmArgs(logger *zap.Logger, rpc string, contractAddr, chainLabel string, mainnetSupported bool) string {
+	if !*unsafeDevMode {
+		// In mainnet / testnet, if either parameter is specified, they must both be specified.
+		if (rpc == "") != (contractAddr == "") {
+			logger.Fatal(fmt.Sprintf("Both --%sContract and --%sRPC must be set or both unset", chainLabel, chainLabel))
+		}
+	} else {
+		// In devnet, if RPC is set but contract is not set, use the deterministic one for tilt.
+		if rpc == "" {
+			if contractAddr != "" {
+				logger.Fatal(fmt.Sprintf("If --%sRPC is not set, --%sContract must not be set", chainLabel, chainLabel))
+			}
+		} else {
+			if contractAddr == "" {
+				contractAddr = devnet.GanacheWormholeContractAddress.Hex()
+			}
+		}
+	}
+	if contractAddr != "" && !mainnetSupported && mainnetMode() {
+		logger.Fatal(fmt.Sprintf("Chain %s not supported in mainnet", chainLabel))
+	}
+	return contractAddr
+}
+
+func mainnetMode() bool {
```

Comment:
if we use `env` above, we can drop this function

Reply:
Created an issue to address this: https://github.com/wormhole-foundation/wormhole/issues/3920